tests/ert/unit_tests/resources/test_run_reservoirsimulator.py

```diff
@@ -0,0 +1,158 @@
+import os
+import stat
+import sys
+import threading
+import time
+from pathlib import Path
+
+import numpy as np
+import pytest
+import resfo
+
+from tests.ert.utils import SOURCE_DIR
+
+from ._import_from_location import import_from_location
+
+# import run_reservoirsimulator from ert/resources/forward_models
+# package-data. This is kept out of the ert package to avoid the
+# overhead of importing ert. This is necessary as these may be invoked as a
+# subprocess on each realization.
+
+
+run_reservoirsimulator = import_from_location(
+    "run_reservoirsimulator",
+    SOURCE_DIR / "src/ert/resources/forward_models/run_reservoirsimulator.py",
+)
+ecl_case_to_data_file = import_from_location(
+    "ecl_case_to_data_file",
+    SOURCE_DIR / "src/ert/resources/forward_models/run_reservoirsimulator.py",
+)
+
+
+@pytest.fixture(name="mocked_eclrun")
+def fixture_mocked_eclrun(use_tmpdir, monkeypatch):
+    """This puts a eclrun binary in path that cannot do anything."""
+    eclrun_bin = Path("bin/eclrun")
+    eclrun_bin.parent.mkdir()
+    eclrun_bin.write_text("", encoding="utf-8")
+    eclrun_bin.chmod(eclrun_bin.stat().st_mode | stat.S_IEXEC)
+    monkeypatch.setenv("PATH", f"bin:{os.environ['PATH']}")
+
+
+def test_unknown_simulator():
```

Comment:
Should this be validated on startup?

Reply:
Should be validated on startup yes, because we must select among supported simulators, it is not possible to pass in arbitrary commands as simulators.